extensions/vault/runtime/src/main/java/io/quarkus/vault/VaultSystemBackendEngine.java

```diff
@@ -0,0 +1,45 @@
+package io.quarkus.vault;
+
+import io.quarkus.vault.runtime.config.VaultRuntimeConfig;
+import io.quarkus.vault.sys.VaultHealth;
+import io.quarkus.vault.sys.VaultHealthStatus;
+import io.quarkus.vault.sys.VaultInit;
+import io.quarkus.vault.sys.VaultSealStatus;
+
+/**
+ * This service provides access to the system backend.
+ *
+ * @see VaultRuntimeConfig
+ */
+public interface VaultSystemBackendEngine {
+
+    /**
+     * Initializes a new Vault.
+     * 
+     * @param secretShares specifies the number of shares to split the master key into.
+     * @param secretThreshold specifies the number of shares required to reconstruct the master key.
+     * @return Vault Init.
+     */
+    VaultInit init(int secretShares, int secretThreshold);
+
+    /**
```

Comment:
@lordofthejars Given what you've discussed with Vincent below, IMHO the java docs have to updated to clarify the difference between `health` and `healthStatus` methods and `VaultHealth` and `VaultHealthStatus` need to be have their names `exchanged` as it is quite confusing IMHO. 
Right now `healthStatus` returns `VaultHealthStatus` which is filled with the real information, while `health` returns `VaultHealth` which only has all that real information available in `VaultHealthStatus` encapsulated  in a status code :-).
Irrespectively of the way the lower level Vault API represents it all, it would be clearer IMHO if we had:
```
// Returns Vault health status code only by using HTTP HEAD requests.
// It is faster than a health() method which uses HTTP GET  to return a complete VaultHealth state
VaultHealthStatus healthStatus(); 

// Returns a complete VaultHealth state.
VaultHealth health(); 
```
Thanks

CC @vsevel

Reply:
I'll update the JavaDoc.